presences/src/main/java/fr/openent/presences/helper/EventQueryHelper.java

```diff
@@ -160,53 +168,54 @@ public static String filterTimes(String startTime, String endTime, JsonArray par
      * This method filter reason (by including several rules between regularized and reasons)
      *
      * @param reasonIds   list of reason ids
-     * @param noReason    no reason wish to not display boolean
+     * @param noAbsenceReason    no reason wish to not display boolean
+     * @param noLatenessReason    no reason wish to not display boolean
      * @param regularized regularized wish to display boolean
      * @param params      JsonArray params where you add the reasonIds, no reason and regularized boolean
      * @return {String}
      */
-    public static String filterReasons(List<String> reasonIds, Boolean noReason, Boolean regularized,
```

Comment:
Nouveau:
`        if ((noReason != null && noReason) && (regularized != null && regularized)) {
            query += "AND (reason_id IS NULL OR (reason_id IS NOT NULL AND counsellor_regularisation = true)";
            if (reasonIds != null && !reasonIds.isEmpty()) {
                query += " AND reason_id IN " + Sql.listPrepared(reasonIds);
                params.addAll(new JsonArray(reasonIds));
            }
            query += ")";

            // else is default condition
        } else {
            // If we want to fetch events WITH reasonId, array reasonIds fetched is not empty
            // (optional if we wish noReason fetched at same time then noReason is TRUE)
            if (reasonIds != null && !reasonIds.isEmpty()) {
                query += " AND ((reason_id IN " + Sql.listPrepared(reasonIds) + ")";

                if (noReason != null && noReason) {
                    query += " OR reason_id IS NULL";
                } else {
                    query += typeIds.contains(EventType.LATENESS.getType().toString())
                            ? (" OR type_id = " + EventType.LATENESS.getType()) : "";
                }
                query += ")";
                params.addAll(new JsonArray(reasonIds));
            }

            // If we want to fetch events with NO reasonId, array reasonIds fetched is empty
            // AND noReason is TRUE
            if (reasonIds != null && reasonIds.isEmpty() && (noReason != null && noReason)) {
                query += " AND (reason_id IS NULL " + (regularized != null ? " OR counsellor_regularisation = " + regularized + "" : "") + ") ";
            }

            if (regularized != null) {
                query += " AND (counsellor_regularisation = " + regularized
                        + " OR type_id != " + EventType.ABSENCE.getType() +") ";
            }
        }`
Ancien:
`// this condition occurs when we want to filter no reason and regularized event at the same time
        if ((noReason != null && noReason) && (regularized != null && regularized)) {
            query += "AND (reason_id IS NULL OR (reason_id IS NOT NULL AND counsellor_regularisation = true)";
            if (reasonIds != null && !reasonIds.isEmpty()) {
                query += " AND reason_id IN " + Sql.listPrepared(reasonIds);
                params.addAll(new JsonArray(reasonIds));
            }
            query += ")";

            // else is default condition
        } else {
            // If we want to fetch events WITH reasonId, array reasonIds fetched is not empty
            // (optional if we wish noReason fetched at same time then noReason is TRUE)
            if (reasonIds != null && !reasonIds.isEmpty()) {
                query += " AND ((reason_id IN " + Sql.listPrepared(reasonIds) + ")";

                if (noReason != null && noReason) {
                    query += " OR reason_id IS NULL";
                } else {
                    query += typeIds.contains(EventType.LATENESS.getType().toString())
                            ? (" OR type_id = " + EventType.LATENESS.getType()) : "";
                }
                query += ")";
                params.addAll(new JsonArray(reasonIds));
            }

            // If we want to fetch events with NO reasonId, array reasonIds fetched is empty
            // AND noReason is TRUE
            if (reasonIds != null && reasonIds.isEmpty() && (noReason != null && noReason)) {
                query += " AND (reason_id IS NULL " + (regularized != null ? " OR counsellor_regularisation = " + regularized + "" : "") + ") ";
            }

            if (regularized != null) {
                query += " AND (counsellor_regularisation = " + regularized
                        + " OR type_id != " + EventType.ABSENCE.getType() +") ";
            }
        }`